### PR TITLE
current.task_log_location() & checkpoint os.listdir fix

### PR DESCRIPTION
**Pull request by talebzeghmi**
* AIP-7056 Metaflow current.log_location() -> path
* checkpoint os.listdir() path exists fix

### CODE REVIEW

File: metaflow/current.py [hunk 1]
```diff
@@ -69,6 +70,31 @@ def is_running_flow(self):
     def flow(self) -> FlowSpec:
         return self._flow
 
+    def log_location(self, log_prefix: str, stream: Optional[str] = None) -> str:
```
Comment (talebzeghmi) [inline metaflow/current.py]: should this be `current.task_log_location()` ??

File: metaflow/current.py [hunk 2]
```diff
@@ -69,6 +70,31 @@ def is_running_flow(self):
...
 
+    def task_log_location(self, log_prefix: str, stream: Optional[str] = None) -> str:
+        """
+        This returns the current Task and Run datastore appropriate location
+        to store logs.
+
+        Args:
+            log_prefix (str): The prefix
+            stream (Optional[str], optional): If None, then the
+                When log_prefix == lightning_logs and stream==None:
```
Comment (tmckay) [inline metaflow/current.py]: This line `When log_prefix == lightning_logs and stream==None:` confuses me and seems like it shouldn't be there or more was meant to be added?

File: metaflow/current.py [hunk 3]
```diff
@@ -69,6 +70,31 @@ def is_running_flow(self):
...
+
+        Examples:
+            Notice that the attempt number is prefixed to the log location path,
+            for example "0." on the first attempt, and on a retry the prefix
+            would be "1."
+
+            >>> current.log_location('lightning_logs')
+                ".metaflow/LightningCLIFlow/4666/start/29503/0.lightning_logs"
+            >>> current.log_location('lightning', 'out')
+                ".metaflow/LightningCLIFlow/4666/_parameters/29503/0.lightning_out.log"
```
Comment (tmckay) [inline metaflow/current.py]: Did you want to change the path to be the step name and not `_parameters`?

**Review by tmckay — APPROVED** [review]

File: metaflow/current.py (hunk 2 above)
Comment (talebzeghmi) [reply metaflow/current.py]: fixed! thank you!

File: metaflow/current.py (hunk 3 above)
Comment (talebzeghmi) [reply metaflow/current.py]: fixed thank you!

**Pull request merged**